### PR TITLE
AC-TF-add-item-feedback

### LINKED ISSUE #9: 9. As a user, I want to be alerted when I’m entering an empty item or an item that's already on my list.
## Summary
Users shouldn't be able to add an empty item to their list, or add the same item twice. If they try to do this, we need to show them an error message that explains the problem. This way, we’ll prevent some clutter in their lists.

## Acceptance criteria
- [ ] Show an error message if the user tries to submit an empty item
- [ ] Show an error message if the user tries to submit a new item that is _identical_ to an existing name. For instance, if the list contains `apples` and the user adds `apples`.
- [ ] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains `apples` and the user adds `aPples` or `apples,` or `a pples`.
- [ ] The user’s original input is saved in the database

##  Notes
- There are a couple of ways to approach this task. If you’re not sure how to begin, _ask your mentors_ for advice!

### PR DESCRIPTION
**Pull request by a-cole55**
## Description

This code allows us to compare the user's input to existing items in the database to ensure that duplicate items aren't added. Will display an error message if the user doesn't enter an item name. Also will display a message if the item name is an exact match or close match to an existing item. 

## Related Issue

closes #9 

## Acceptance Criteria

- [x] Show an error message if the user tries to submit an empty item
- [x] Show an error message if the user tries to submit a new item that is identical to an existing name. For instance, if the list contains apples and the user adds apples.
- [x] Show an error message if the user tries to submit a new item that matches an existing name with punctuation and casing normalized. For instance, if the list contains apples and the user adds aPples or apples, or a pples.
- [x]  The user’s original input is saved in the database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |

### Before

before

### After

after-exact-match

after-normalized-input

## Testing Steps / QA Criteria

1.  git pull
2.  git checkout AC-TF-add-item-feedback
3.  Navigate to Add Item Page
4.  Type item names

### CODE REVIEW

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -25,18 +26,25 @@ export function AddItem() {
 		setFormData((formData) => ({ ...formData, [name]: value }));
 	};
 
+	const itemNames = data.map((item) =>
+		item.name.toLowerCase().replace(/[^a-z0-9]/gi, ''),
```
Comment (mariahschock) [inline src/views/AddItem.jsx:31]: I'm not very familiar with regex, so the .replace(/[^a-z0-9]/gi, '') was really cool to see! Really neat way to handle case insensitivity.

File: src/views/AddItem.jsx [hunk 2]
```diff
@@ -25,18 +26,25 @@ export function AddItem() {
...
 	// if item name provided, adds item to user's list and gives user success feedback
 	const handleSubmit = (e) => {
 		e.preventDefault();
 		if (itemName === '') {
-			alert('Please enter the name of your item.');
+			setMessage('Please enter the name of your item');
+		} else if (
+			itemNames.includes(itemName.toLowerCase().replace(/[^a-z0-9]/gi, ''))
+		) {
+			setMessage(`You already have ${itemName} on your list`);
```
Comment (mariahschock) [inline src/views/AddItem.jsx:42]: This looks really great a feels more user friendly than the alert.

**Review by mariahschock — APPROVED** [review]
Great work you two! I love the new error messages, and the red definitely draws the eye.

File: src/views/AddItem.jsx (hunk 1 above)
Comment (wheninseattle) [inline src/views/AddItem.jsx:31]: Nice use of regex!

File: src/views/AddItem.jsx (hunk 2 above)
Comment (wheninseattle) [reply src/views/AddItem.jsx:42]: This is great! If we have time, I'd love to see this implement elsewhere in the project. One todo for later: right now, the text is red regardless of the feedback. At some point we may want to add a second argument to the alert to give it a type eg) `setMessage({type: 'error', msg: 'You already...})` or `setMessage({type: 'success', msg:' ${itemName} added to list'})`

**Review by wheninseattle — APPROVED** [review]
Awesome job you two! Loved the use of regex and the transition to creating an element for alerts. I left a couple of comments about potential things to tweak in the coming weeks if we have time, but overall great work! 🥳🐨🦊

File: src/views/AddItem.jsx (hunk 2 above)
Comment (tmflower) [reply src/views/AddItem.jsx:42]: Red inline styling is just temporary, lol; I'm sure we'll remove that and handle styling with css in our final stage.

File: src/views/AddItem.jsx (hunk 1 above)
Comment (veeteeran) [reply src/views/AddItem.jsx:31]: The replace method is very powerful combined with the regex! Lots of ways to do the matching! This is as good as any `[^a-z0-9]` 🎉

File: src/views/AddItem.jsx (hunk 2 above)
Comment (veeteeran) [reply src/views/AddItem.jsx:42]: Good comment about the text color and agree with you both here. Red generally indicates warning or error. Prioritizing as a todo for later is a great call!

**Review by veeteeran — APPROVED** [review]
Great work! Clear messaging to the user when they succeed in adding items or why they can't. The string sanitization and matching are spot on!!!

**Pull request merged**